### PR TITLE
Restore validation reporting

### LINKED ISSUE #80: Validation reporting broken
Validation errors are no longer reported, even when running with VKB_VALIDATION_LAYERS enabled.

#60 modified the code to add `VkDebugReportCallbackCreateInfoEXT` structure to the `pNext` for the instance create info, adding support for reporting validation errors during instance creation.  However, it also _removed_ the calls to `vkCreateDebugReportCallbackEXT`, presumably in the belief that the instance creation change was a replacement for this logic.  It is not.  While the structure chaining during instance creation gives the driver enough to report validation errors in creation, it provides no way to populate `VkDebugReportCallbackEXT debug_report_callback` for later manipulation of the reporting.

### PR DESCRIPTION
**Pull request by jherico**
## Description

Closes #80 

## General Checklist:

Please ensure the following points are checked:

- [X] My code follows the [coding style](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/CONTRIBUTING.md#Code-Style)
- [X] I have reviewed file [licenses](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/CONTRIBUTING.md#Copyright-Notice-and-License-Template)
- [X] I have commented any added functions (in line with Doxygen)
- [X] I have commented any code that could be hard to understand
- [X] My changes do not add any new compiler warnings
- [X] My changes do not add any new validation layer errors or warnings
- [X] I have used existing framework/helper functions where possible
- [X] My changes build and run on Windows, Linux, macOS and Android. Otherwise I have [documented any exceptions](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/CONTRIBUTING.md#General-Requirements)

### CODE REVIEW

File: framework/core/instance.cpp [hunk 1]
```diff
@@ -248,6 +248,14 @@ Instance::Instance(const std::string &              application_name,
 
 	volkLoadInstance(handle);
 
+#if defined(VKB_DEBUG) || defined(VKB_VALIDATION_LAYERS)
+	result = vkCreateDebugReportCallbackEXT(handle, &debug_report_create_info, nullptr, &debug_report_callback);
+	if (result != VK_SUCCESS)
+	{
+		throw VulkanException(result, "Could not create Vulkan instance");
```
Comment (SaschaWillems) [inline framework/core/instance.cpp]: Can you change the error message to hint at the debug report callback creation failure?

**Review by SaschaWillems — CHANGES_REQUESTED** [review]

File: framework/core/instance.cpp (hunk 1 above)
Comment (TomAtkinsonArm) [reply framework/core/instance.cpp]: Looks good after this change :)

**Review by ZandroFargnoli — DISMISSED** [review]
Well noticed @jherico @SaschaWillems ! Changes LGTM after the suggestion Sascha made

File: framework/core/instance.cpp (hunk 1 above)
Comment (jherico) [reply framework/core/instance.cpp]: fixed.

**Review by ZandroFargnoli — APPROVED** [review]
LGTM!

**Comment by TomAtkinsonArm** [comment]
All good

**Review by TomAtkinsonArm — APPROVED** [review]

**Pull request merged**